src/foundry/client/pixi/webgl/shaders/fragment-channel-mixin.d.mts

```diff
@@ -0,0 +1,56 @@
+import type { AnyObject, Mixin, ShapeWithIndexSignature } from "../../../../../types/utils.d.mts";
+
+export {};
+
+declare abstract class AnyAdaptiveFragmentChannel extends AdaptiveFragmentChannel {
+  constructor(arg0: never, ...args: never[]);
+}
+
+declare class AdaptiveFragmentChannel {
+  /** @privateRemarks All mixin classses should accept anything for its constructor. */
+  constructor(...args: any[]);
+
+  /**
+   * The fragment shader which renders this filter.
+   * A subclass of AdaptiveFragmentChannelMixin must implement the fragmentShader static field.
+   * @defaultValue `null`
+   */
+  static adaptiveFragmentShader: ((channel: AdaptiveFragmentChannel.Channel) => string) | null;
+
+  /**
+   * A factory method for creating the filter using its defined default values
+   * @param options - Options which affect filter construction
+   */
+  static create<T extends AnyObject>({
+    /**
+     * A color channel to target for masking.
+     * @defaultValue `r`
+     */
+    channel,
+    /**
+     * Initial uniforms provided to the filter
+     */
+    ...uniforms
+  }?: ShapeWithIndexSignature<
+    T,
+    AdaptiveFragmentChannel.ConcreteCreateOptions,
+    string,
+    AbstractBaseShader.UniformValue
+  >): PIXI.Shader | PIXI.Filter;
+}
+
+declare global {
+  namespace AdaptiveFragmentChannel {
+    type AnyConstructor = typeof AnyAdaptiveFragmentChannel;
+
+    type Channel = "r" | "g" | "b";
+
+    interface ConcreteCreateOptions {
+      channel: AdaptiveFragmentChannel.Channel | undefined;
```

Comment:
probably should be `channel?: AdaptiveFragmentChannel.Channel | undefined;` unless there's a specific reason why `{}` isn't okay but `{ channel: undefined }` is.

Reply:
done, I think I had it InexactPartial'd at some point and then changed that.